0000-product.md

```diff
@@ -0,0 +1,504 @@
+product_id, a product_namespace, an owner, and properties (a repeated key-value
+field)._
+
+
+# Motivation
+[motivation]: #motivation
+
+The Grid Product implementation is designed for sharing product master data
+between participants. Product is a near universal concept within supply chain
+solutions and would naturally be one of the highest areas of re-use across Grid
+applications.
+
+The design will address use cases including:
+
+- Sharing of Product master data across a network
+- Including Product in other business transactions (track and trace events,
+  purchases, sales, etc)
+- Enriching UX experiences by including additional attribution of a Product such
+  as names or descriptions
+
+It is also useful to use product information as auxiliary data in other supply
+chain solutions.  For example, in track and trace, the location and temperature
+of an item (an instance of a product) is stored.  When presenting this
+information later to a human user, it is useful to combine it with information
+about the product, such as the product’s name and description.  Therefore,
+Product becomes a reusable and important component within Grid.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Entities
+
+A **__product__** is an archetype of an item that is transacted, traded, or
+referenced in a supply chain.  Each product has a **__product_namespace__**.
+This RFC defines a single product_namespace for GS1; a product in the GS1
+product_namespace is called a GS1 product.  Note that this design supports
+extending additional product namespaces in the future.
+
+A product is referenced using a **product_id**.  For GS1 products, the
+product_id is a Global Trade Item Number (GTIN), which is part of the GS1
+specifications.
+
+Each product has an **__owner__**, which is the identifier of the organization
+responsible for maintaining the product.  An **__agent__** acts on behalf of an
+organization, and all permissions are defined in terms of the actions agents can
+perform.  Organizations and agents are defined and managed by Pike (another
+component of Grid).
+
+A product has one or more **properties**.  Properties are described in the Grid
+Primitives RFC.  A *property namespace* contains multiple *property schemas*.  A
+property schema associates a name (such as “length”) with a data type (such as
+integer).  GS1 products may only include properties defined in the GS1 product
+property namespace.
+
+## Transactions
+
+Products are managed by submitting transactions to Hyperledger Grid, which will
+process them with the Grid Product smart contract. The following transactions
+are supported:
+
+* ProductCreate - create a Product and store it in state.  
+* ProductUpdate - update (replace) the properties of a Product already in
+  state.
+* ProductDelete - remove a Product from state.
+
+*Disclaimer: ProductDelete does not remove the record of existance of a product.
```

Comment:
This implies some behaviors of the underlying state model that may not hold true. I'd suggest getting rid of this disclaimer, or just stating that deletion is a potentially hazardous operation and needs to be done with care.

Reply:
Keeping the disclaimer, changed it to your suggestion.